### PR TITLE
[Feature] New Parser and Analyzer support SetUserPropertyStmt/ShowUserPropertyStmt

### LINKED ISSUE #7958: New Parser and Analyzer support SetUserPropertyStmt/ShowUserPropertyStmt （Easy）
# Required work and output

1.  Use antlr4 to complete parse statements. The generated statement must follow the original data structure, and should be compatible with the old statement.
    
2.  In the original syntax file  **sql_parser.cup**, the new syntax needs to be compatible with the syntax of the old version, using the syntax file sql_parser.cup of the old version as the reference standard.
    
3.  Remove the logic of analytic functions in the statement and move the relevant logic to the new Analyzer framework. The new Analyzer is in the SQL/Analyzer directory.
    
4.  If the code related to Privilege Check exists in the original analyzer, move the code related to privilege control into PrivilegeChecker.
    
5.  When submitting the PR, please make sure there is unit test coverage for related changes.
    

# Task
 - [ ] SetUserPropertyStmt/ShowUserPropertyStmt

# How to participate?

1.  Pick an issue you are interested in, click the issue link and leave a comment under the issue to sign up.  **One person can only choose one issue and the first person who picks the issue will be assigned.**
    
2.  **After the issue is assigned, you should complete the task in 14 days.** If you are unable to complete it within the specified time, the task will be withdrawn and assigned to others.
    
3.  Participants are required to produce two pieces of content (forms including but not limited to mind maps, study notes, blogs, and tutorials) in the course of the activity, each of which should be at least 200 words.
    
4.  Got questions? Join our  [Slack channel](https://join.slack.com/t/starrocks/shared_invite/zt-192zeqlc7-9MAvqAXNcveW_l5Pqr8r2g)  and find channel:  **#sql-planner**.
    

# Before you start

1.  Read  [Contributing.md](https://github.com/StarRocks/starrocks/blob/main/CONTRIBUTING.md)  carefully to understand how to submit a pull request and how to configure BE and FE development environment.
    
2.  Sign the  [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks)  before you make a pull request.
    
3.  To learn more about StarRocks, please refer to  [StarRocks documentation](https://github.com/StarRocks/docs/blob/main/TOC.md).
    

# Pull request requirements

1.  Create a pull request (PR) after you complete the task.
* PR title format: [SR SQL Planner Summer Camp] + PR title
2. After submitting PR, you can comment under the pinned Issue [SR SQL Planner Summer Camp].
* Comment format: "I have finished Issue #" + PR link

### PR DESCRIPTION
**Pull request by xuzifu666**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7958

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java [hunk 1]
```diff
@@ -97,6 +99,16 @@ public Void visitAdminShowReplicaDistributionStatement(AdminShowReplicaDistribut
...
         }
 
+        @Override
+        public Void visitShowUserPropertyStmt(ShowUserPropertyStmt statement, ConnectContext session) {
+            return null;
+        }
+
+        @Override
+        public Void visitSetUserPropertyStmt(SetUserPropertyStmt statement, ConnectContext context) {
+            return null;
```
Comment (xlwh) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java:109]: You need to implement logic like: SetUserPropertyStmt.analyze()

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java [hunk 2]
```diff
@@ -97,6 +99,16 @@ public Void visitAdminShowReplicaDistributionStatement(AdminShowReplicaDistribut
             return null;
         }
 
+        @Override
+        public Void visitShowUserPropertyStmt(ShowUserPropertyStmt statement, ConnectContext session) {
+            return null;
```
Comment (xlwh) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java:104]: You need to implement logic like: ShowUserPropertyStmt.analyze()

**Review by xlwh — COMMENTED** [review]
1. You need to migrate the analyze logic to new parser.
2. You need migrate the privilege checker logic to the PrivilegeChecker.java
3. Add some unit test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 37 / 64 (57.81%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|2|00.00%|[325, 329]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|16|00.00%|[220, 222, 223, 224, 225, 227, 232, 233, 236, 238, 239, 240, 241, 242, 243, 245]|
|:large_blue_circle:|com/starrocks/analysis/SetUserPropertyStmt.java|3|5|60.00%|[50, 89]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SetUserPropertyAnalyzer.java|5|7|71.43%|[10, 17]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowUserPropertyAnalyzer.java|5|7|71.43%|[9, 14]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|13|15|86.67%|[1853, 1854]|
|:large_blue_circle:|com/starrocks/analysis/ShowUserPropertyStmt.java|7|8|87.50%|[125]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|4|4|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SetUserPropertyAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        if (Strings.isNullOrEmpty(user)) {
+            // If param 'user' is not set, use the login user name.
+            // The login user name is full-qualified with cluster name.
+            statement.setUser(ConnectContext.get().getQualifiedUser());
+        } else {
+            // If param 'user' is set, check if it need to be full-qualified
+            if (!user.equals(Auth.ROOT_USER) && !user.equals(Auth.ADMIN_USER)) {
+                statement.setUser(ClusterNamespace.getFullName(user));
+            }
+        }
```
[the remark this answers is not in the document]
Comment (xuzifu666) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SetUserPropertyAnalyzer.java:23]: this logic had added to PrivilegeChecker

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 46 / 64 (71.88%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|2|00.00%|[325, 329]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|8|16|50.00%|[222, 225, 227, 233, 239, 240, 243, 245]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SetUserPropertyAnalyzer.java|5|7|71.43%|[10, 17]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowUserPropertyAnalyzer.java|5|7|71.43%|[9, 14]|
|:large_blue_circle:|com/starrocks/analysis/SetUserPropertyStmt.java|4|5|80.00%|[89]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|13|15|86.67%|[1853, 1854]|
|:large_blue_circle:|com/starrocks/analysis/ShowUserPropertyStmt.java|7|8|87.50%|[125]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|4|4|100.00%|[]|

**Review by xlwh — APPROVED** [review]
LGTM

**Review by imay — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**